javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -80,499 +122,608 @@ module TaintedPath {
+          operand = startsWith.getBaseString() and
+          polarity = startsWith.getPolarity() and
+          if substring = "" then negatable = true else negatable = false
+        ) // !x.startsWith("/home") does not guarantee that x is not absolute
+      }
 
-  /**
-   * A call to `path.isAbsolute` as a sanitizer for relative paths in true branch,
-   * and a sanitizer for absolute paths in the false branch.
-   */
-  class IsAbsoluteSanitizer extends BarrierGuardNode {
-    DataFlow::Node operand;
-    boolean polarity;
-    boolean negatable;
-
-    IsAbsoluteSanitizer() {
-      exists(DataFlow::CallNode call | this = call |
-        call = NodeJSLib::Path::moduleMember("isAbsolute").getACall() and
-        operand = call.getArgument(0) and
-        polarity = true and
-        negatable = true
-      )
-      or
-      exists(StringOps::StartsWith startsWith, string substring | this = startsWith |
-        startsWith.getSubstring().getStringValue() = "/" + substring and
-        operand = startsWith.getBaseString() and
-        polarity = startsWith.getPolarity() and
-        if substring = "" then negatable = true else negatable = false
-      ) // !x.startsWith("/home") does not guarantee that x is not absolute
-    }
-
-    override predicate blocks(boolean outcome, Expr e, DataFlow::FlowLabel label) {
-      e = operand.asExpr() and
-      exists(Label::PosixPath posixPath | posixPath = label |
-        outcome = polarity and posixPath.isRelative()
+      override predicate blocks(boolean outcome, Expr e, DataFlow::FlowLabel label) {
+        e = operand.asExpr() and
+        outcome = polarity and
+        label.(Path::PathLabel).isRelative()
         or
         negatable = true and
         outcome = polarity.booleanNot() and
-        posixPath.isAbsolute()
-      )
-    }
-  }
-
-  /**
-   * An expression of form `x.includes("..")` or similar.
-   */
-  class ContainsDotDotSanitizer extends BarrierGuardNode instanceof StringOps::Includes {
-    ContainsDotDotSanitizer() { isDotDotSlashPrefix(super.getSubstring()) }
-
-    override predicate blocks(boolean outcome, Expr e, DataFlow::FlowLabel label) {
-      e = super.getBaseString().asExpr() and
-      outcome = super.getPolarity().booleanNot() and
-      label.(Label::PosixPath).canContainDotDotSlash() // can still be bypassed by normalized absolute path
-    }
-  }
-
-  /**
-   * An expression of form `x.matches(/\.\./)` or similar.
-   */
-  class ContainsDotDotRegExpSanitizer extends BarrierGuardNode instanceof StringOps::RegExpTest {
-    ContainsDotDotRegExpSanitizer() { super.getRegExp().getAMatchedString() = [".", "..", "../"] }
-
-    override predicate blocks(boolean outcome, Expr e, DataFlow::FlowLabel label) {
-      e = super.getStringOperand().asExpr() and
-      outcome = super.getPolarity().booleanNot() and
-      label.(Label::PosixPath).canContainDotDotSlash() // can still be bypassed by normalized absolute path
+        label.(Path::PathLabel).isAbsolute()
```

Comment:
You forgot to use `e` in the second part of this `blocks` predicate.  
(Code-scanning have a warning about it).